src/flake8_tidy_imports/__init__.py

```diff
@@ -166,6 +183,73 @@ def _is_module_banned(self, module_name: str) -> tuple[bool, str]:
 
         return False, ""
 
+    def _is_idiom_banned(
+        self,
+        imported_module: str,
+        imported_alias: str,
+    ) -> tuple[bool, str]:
+        for module_name, metadata in self.import_idioms.items():
```

Comment:
iterating through all idioms is not going to be fast. We need a datastructure with an O(1) lookup here, otherwise the plugin may slow down linting with large numbers of idioms.